jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/internal/HttpReceiverOverHTTP.java

```diff
@@ -66,7 +66,7 @@ public HttpReceiverOverHTTP(HttpChannelOverHTTP channel)
             parser.setHeaderCacheSize(httpTransport.getHeaderCacheSize());
             parser.setHeaderCacheCaseSensitive(httpTransport.isHeaderCacheCaseSensitive());
         }
-        retainableByteBufferPool = httpClient.getByteBufferPool().asRetainableByteBufferPool();
+        retainableByteBufferPool = httpClient.getRetainableByteBufferPool();
```

Comment:
Why is the `networkBuffer` not released in the `failAndClose` method?

Reply:
Because `failAndClose()` is called by parser callbacks. After these callbacks return, we are back into `parseAndFill()` where we either continue or stop, but in both cases the `networkBuffer` is taken care of.

Put it in another way, the only place where the `networkBuffer` is acquired is within `parseAndFill()`, and there it is also where it is released.